### DISCUSSION TITLE
EMP should create disturbances in logic circuits

### DISCUSSION THREAD

**Discussion by mygamingaccount** [FakeFishGames/Barotrauma#10216] [Ideas and suggestions]
When the ship is hit by an EMP from a nuclear blast, Black Moloch, or an EMP grenade, have each wire transmit a `1` signal to whatever it is connected to.
Perhaps reduce the chance of it happening to 5%-10%. The exact chance should come after testing I suppose.

**Comment by MapleWheels** [comment]
> When the ship is hit by an EMP from a nuclear blast, Black Moloch, or an EMP grenade, have each wire transmit a `1` signal to whatever it is connected to. Perhaps reduce the chance of it happening to 5%-10%. The exact chance should come after testing I suppose.

I agree this would be a very nice touch but it could render many subs unusable permanently as almost anything that exists now isn't made for it. Asking sub-makers to add ECC to circuits, many of whom are not proficient in circuit design, would be a disaster. There would need to be a change to allow graceful resets. 

Don't get me wrong, it can be done, would be a nice effect, but it's a bit more complicated than it seems from the end-user's perspective.

**Reply by mygamingaccount** [reply]
That's one of the benefits of Early Access, you can break compatibility with User Generated Content at any time.

The results of an EMP strike:
- Humpback/Typhon2 battery recharge rate changed
- Battery relays may get swapped around
- Cannons randomly fire
- Doors randomly open

But nothing catastrophic as far as I can tell

**Reply by MapleWheels** [reply]
> That's one of the benefits of Early Access, you can break compatibility with User Generated Content at any time.
> 
> The results of an EMP strike:
> 
> * Humpback/Typhon2 battery recharge rate changed
> * Battery relays may get swapped around
> * Cannons randomly fire
> * Doors randomly open
> 
> But nothing catastrophic as far as I can tell

No, not completely. You have to realize that the community is only willing to fix so much and deal with so much complexity before they will stop. It should only be done after a certain amount of consideration and if you can justify it.

**Reply by MateusJK15** [reply]
> That's one of the benefits of Early Access, you can break compatibility with User Generated Content at any time.

Only if it's REALLY necessary. In this case it's not, the devs just have to make the EMP effect on the submarine components be temporary, forcing them to send 1 for X seconds without permanently changing its settings, like ballast flora does.

**Reply by mygamingaccount** [reply]
That's an interesting suggestion. I was thinking of an instantaneous event, not one that keeps sending signals over time.

Some effects of signals are permanent, like I mentioned, if you had a wire going to your battery set_charge_rate pin, an EMP could change it.

**Comment by wthrajat** [comment]
mind me gentlemen, but i was wondering why did I get the notification of this discussion? I do not remember subscribing the discussion. Can anyone enlighten me?

**Reply by neruthes** [reply]
Same problem here

**Comment by Attikittyalt1** [comment]
Maybe have a toggle on each component for whether it's affected by EMP (defaults to on), so that sub-builders could keep core-circuits always working. Then if a wire sends a signal generated by EMP to a component as an input, it would only affect it if the toggle is on.
